### PR TITLE
feat(templates/deno): add triple-slash reference for client code

### PR DESCRIPTION
**Pull request by justjavac**
- `/// <reference no-default-lib="true" />` can prevent developers from writing Deno code on the client side.

- `<reference lib="dom" />` can fix using document in Deno code. Without this line of code, vscode produces a warning:

  > Cannot find name 'document'. Do you need to change your target library? Try changing the 'lib' compiler option to include 'dom'. **deno-ts(2584)**

### CODE REVIEW

File: templates/deno/app/entry.client.tsx [hunk 1]
```diff
@@ -1,3 +1,6 @@
+/// <reference no-default-lib="true" />
+/// <reference lib="dom" />
```
Comment (MichaelDeBoey) [inline templates/deno/app/entry.client.tsx:2]: I think this should be in a `tsconfig.json` instead? 🤔

CC/ @pcattori

Comment (MichaelDeBoey) [reply templates/deno/app/entry.client.tsx:2]: Other templates have it in its `tsconfig.json`.

I know @pcattori removed the file on purpose, but I can't remember the reasoning behind it.

**Comment by MichaelDeBoey** [comment]
@justjavac Template changes go against `main`.
Could you please rebase your branch onto latest `main` & set base branch to `main` as wel?

**Comment by justjavac** [comment]
@MichaelDeBoey I opened a new pr #3449

**Pull request closed without merging**